index.js

```diff
@@ -140,8 +160,67 @@ class Emittery {
 
 		return count;
 	}
+
+	bindMethods(target, methodNames) {
+		if (typeof target !== 'object' || target === null) {
+			throw new TypeError('target must be an object');
+		}
+		methodNames = defaultMethodNamesOrAssert(methodNames);
+		methodNames.forEach(methodName => {
+			if (target[methodName] !== undefined) {
+				throw new Error(`field ${methodName} already exists on targer`);
+			}
+		});
+
+		for (const methodName of methodNames) {
+			Object.defineProperty(target, methodName, {
+				enumerable: false,
+				value: this[methodName].bind(this)
+			});
+		}
+	}
+
+	static mixin(emitteryPropertyName, methodNames) {
```

Comment:
Is this following the latest decorator spec? https://github.com/tc39/proposal-decorators

Reply:
It is not a spec, it is a proposal. There is no implementation which can handle decorators implemented in such a way yet. I followed the typescript implementation of decorators.

Reply:
Alright. But it needs to be clearly documented in the type and readme that it follows the TypeScript implementation.

Reply:
I described it only for typescript part of readme so far(as using it from JS is weird anyway atm).